cmake/zephyr_module.cmake

```diff
@@ -0,0 +1,83 @@
+# This cmake file provides functionality to import additional out-of-tree, OoT
+# CMakeLists.txt and Kconfig files into Zephyr build system.
+# It uses -DZEPHYR_MODULES=<oot-path-to-module>[;<additional-oot-module(s)>]
+# given to CMake for a list of folders to search.
+# It looks for: <oot-module>/zephyr/module.yml or
+#               <oot-module>/zephyr/CMakeLists.txt
+# to load the oot-module into Zephyr build system.
+# If west is available, it uses `west list` to obtain a list of projects to
+# search for zephyr/module.yml
+
+# Clear zephyr modules property
+set_property(GLOBAL PROPERTY ZEPHYR_MODULES_NAME "")
+set_property(GLOBAL PROPERTY ZEPHYR_MODULES_DIR  "")
+
+if(ZEPHYR_MODULES)
+  set(west_project_list ${ZEPHYR_MODULES})
+else()
+  ## Use `west list` to fetch all west handled projects.
+  execute_process(
+    COMMAND
+    west list --format={abspath}\;
+    OUTPUT_VARIABLE
+    west_project_list
+  )
+  string(REGEX REPLACE "[\r\n]+" ";" west_project_list "${west_project_list}")
+endif()
+
+if(ZEPHYR_EXTRA_MODULES)
+  list(APPEND west_project_list ${ZEPHYR_EXTRA_MODULES})
+endif()
+
+# Clear the Kconfig.modules generated file in case modules has been removed.
+# The Kconfig.modules contains a list of additional Kconfig files to be sourced
+# based upon <module>/zephyr/module.yml files.
+file(WRITE ${PROJECT_BINARY_DIR}/Kconfig.modules
+  "# NOTE: THIS FILE IS AUTOGENERATED BY CMAKE\n"
+  )
+# For each west managed project, determine if the project is a zephyr module.
+foreach(module ${west_project_list})
+  set(cmake_subdir "zephyr")
+  if(${ZEPHYR_BASE} STREQUAL ${module})
+    # Ignore Zephyr project to avoid potential invalid looping
+  elseif(EXISTS "${module}/zephyr/module.yml")
+    set(kconfig_osource "osource \"${module}/zephyr/Kconfig\"\n")
+    execute_process(
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/scripts/yaml_to_cmake.py
+      -f "${module}/zephyr/module.yml"
+      -s "build"
+      OUTPUT_VARIABLE west_module
+      )
+
+    foreach(key_value ${west_module})
+      if(${key_value} MATCHES "^cmake=")
+        string(REGEX REPLACE "^cmake=" "" cmake_subdir ${key_value})
+      endif()
+      if(${key_value} MATCHES "^kconfig=")
+        string(
+          REGEX REPLACE
+          "^kconfig="
+          "osource \"${module}/"
+          kconfig_osource
+          ${key_value} "\"\n"
+          )
+      endif()
+    endforeach()
+
+    get_filename_component(module_name ${module} NAME)
+    set_property(GLOBAL APPEND PROPERTY ZEPHYR_MODULES_NAME ${module_name})
+    set_property(GLOBAL APPEND PROPERTY ZEPHYR_MODULES_DIR ${module}/${cmake_subdir})
+  elseif(EXISTS "${module}/${cmake_subdir}/CMakeLists.txt")
+    set(kconfig_osource "${module}/zephyr/Kconfig\n")
```

Comment:
shouldn't it be 

`osource ${module}/zephyr/Kconfig\n`

?

Reply:
correct